examples/oft-hyperliquid/deploy/MyHyperliquidOFT.ts

```diff
@@ -0,0 +1,103 @@
+import assert from 'assert'
+
+import { Wallet, providers } from 'ethers'
+import { HttpNetworkConfig } from 'hardhat/types'
+import { type DeployFunction } from 'hardhat-deploy/types'
+
+import { useBigBlock, useSmallBlock, writeUpdatedNativeSpots } from '@layerzerolabs/oft-hyperliquid-evm'
+
+const contractName_oft = 'MyHyperLiquidOFT'
+const tokenSymbol = ''
+const nativeSpotName = ''
+
+const deploy: DeployFunction = async (hre) => {
+    assert(tokenSymbol != '', 'tokenSymbol needs to be set in deploy/MyHyperLiquidOFT.ts')
+    assert(nativeSpotName != '', 'nativeSpotName needs to be set in deploy/MyHyperLiquidOFT.ts')
+
+    const { getNamedAccounts, deployments } = hre
+
+    const { deploy } = deployments
+    const { deployer } = await getNamedAccounts()
+    assert(deployer, 'Missing named deployer account')
+
+    const privateKey = process.env.PRIVATE_KEY_HYPERLIQUID
+    assert(privateKey, 'PRIVATE_KEY_HYPERLIQUID is not set in .env file')
+
+    // Get logger from hardhat flag --log-level
+    const loglevel = hre.hardhatArguments.verbose ? 'debug' : 'info'
+
+    const wallet = new Wallet(privateKey)
+    const isTestnet = hre.network.name === 'hyperliquid-testnet'
+
+    const networkName = hre.network.name
+    console.log(`Network: ${networkName}`)
+    console.log(`Deployer: ${deployer}`)
+
+    assert(
+        networkName === 'hyperliquid-testnet' || networkName === 'hyperliquid-mainnet',
+        'This deploys to hyperliquid networks'
```

Comment:
A better message might be `Deployment is only supported on HyperLiquid networks`